es-codec.ts

```diff
@@ -190,6 +193,42 @@ function decodeDate(buffer : ArrayBuffer, cursor : Cursor) {
     return new Date(view.getFloat64(0))
 }
 
+const REGEX_GLOBAL_FLAG      = 0b00001
+const REGEX_IGNORE_CASE_FLAG = 0b00010
+const REGEX_MULTILINE_FLAG   = 0b00100
+const REGEX_UNICODE_FLAG     = 0b01000
+const REGEX_STICKY_FLAG      = 0b10000
+function encodeRegex(regex : RegExp) {
+    const lastByte = 0
+        | (regex.global     ? REGEX_GLOBAL_FLAG      : 0)
+        | (regex.ignoreCase ? REGEX_IGNORE_CASE_FLAG : 0)
+        | (regex.multiline  ? REGEX_MULTILINE_FLAG   : 0)
+        | (regex.unicode    ? REGEX_UNICODE_FLAG     : 0)
+        | (regex.sticky     ? REGEX_STICKY_FLAG      : 0)
+    const numberOfPositiveFlags = 0
+        + (+regex.global)
+        + (+regex.ignoreCase)
+        + (+regex.multiline)
+        + (+regex.unicode)
+        + (+regex.sticky)
+    // regexString  = /blah/gi
+    // regexContent =  blah     // which is regexString.slice(1,-(numberOfPositiveFlags+1)
+    const encodedBuffer = new TextEncoder().encode(regex.toString().slice(1,-(numberOfPositiveFlags+1))+String.fromCharCode(lastByte)).buffer
+    return concatArrayBuffers(Uint8Array.of(REGEX).buffer, encodeVarint(encodedBuffer.byteLength).buffer, encodedBuffer)
+}
```

Comment:
I think that's really clean, it also means the code doesn't need to be updated if new flags are added.

Reply:
I imagine just concating them will be a problem though, as there needs to be a way to tell the difference between something like /a/gi and /ag/i

I think any delimiter will be fine, so "/" seems like it would work. Tomorrow I can update the decoder if you don't beat me to it.

Reply:
> I imagine just concating them will be a problem though, as there needs to be a way to tell the difference between something like /a/gi and /ag/i

That won't be an issue. When decoding, the first call to decode will yield the source, and the second the flags.

Reply:
Oh, now I see that its using the full encode string, so the string tag will be there.